### PR TITLE
#254 3DB Build list

### PR DESCRIPTION
**Pull request by 3rdvision**
| - | - |
| --- | --- |
| Issue | https://github.com/ripe-tech/ripe-white/issues/250 |
| Dependencies | - |
| Decisions | - |
| Animated GIF | - |

### CODE REVIEW

**Comment by 3rdvision** [comment]
@gcandal I don't have permissions assign you. Mentioning you in this comment instead.

File: src/js/api/build.js [hunk 1]
```diff
@@ -11,6 +11,50 @@ if (
...
+ * - 'sort' - List of arguments to sort the results by and which direction
+ * to sort them in (eg: 'id:ascending') would sort by the id attribute in ascending order,
+ * while (eg: 'id:descending')] would do it in descending order.
+ * - 'skip' - The number of the first record to retrieve from the results.
+ * - 'limit' - The number of results to retrieve.
+ * @param {Function} callback Function with the result of the request.
+ * @param {Function} callback Function with the result of the request.
+ * @returns {XMLHttpRequest} The XMLHttpRequest instance of the API request.
+ */
+ripe.Ripe.prototype.getBuilds = function(options, callback) {
```
Comment (gcandal) [inline src/js/api/build.js:30]: @3rdvision actually we don't support all these options. Unfortunately, the build listing endpoint is much dumber (see https://github.com/ripe-tech/ripe-core/blob/89e4b47edf05120e33e59c8c86bb20568979eb80/src/ripe_core/controllers/api/build.py#L20). So we need to remove that from here.

**Comment by 3rdvision** [comment]
@gcandal

File: src/python/ripe_demo/static/js/main.js [hunk 2]
```diff
@@ -4,7 +4,7 @@
  */
 var FACES = ["side", "top", "front"];
 
-window.onload = function() {
```
Comment (gcandal) [inline src/python/ripe_demo/static/js/main.js:7]: @3rdvision something smells fishy. Make sure you have the latest dependencies, this seems like an old prettier config.

File: src/js/api/build.js [hunk 3]
```diff
@@ -11,6 +11,26 @@ if (
     var ripe = base.ripe;
 }
 
+/**
+ * Gets the existing builds
+ *
+ * @param {Function} callback Function with the result of the request.
+ * @returns {XMLHttpRequest} The XMLHttpRequest instance of the API request.
+ */
+ripe.Ripe.prototype.getBuilds = function(callback) {
```
Comment (gcandal) [inline src/js/api/build.js:20]: Add the promise version too (`getBuildsP`).

File: src/python/ripe_demo/static/js/main.js (hunk 2 above)
Comment (3rdvision) [reply src/python/ripe_demo/static/js/main.js:7]: Even after removing and reinstalling modules and even upgrading, it formats this way after `npm run pretty`... wierd! 🐟 
Reverting to original.

**Comment by gcandal** [comment]
@joamag please take a look at this.

**Comment by joamag** [comment]
@3rdvision I've just re-added permission to write to this repository. We've just changed the permissions scheme and some of you may have lost access to some repos. Please ping me when you don't have access to a repo and I'll re-add permissions.

**Pull request merged**